crates/core/component/shielded-pool/src/component/shielded_pool.rs

```diff
@@ -20,17 +21,16 @@ pub struct ShieldedPool {}
 
 #[async_trait]
 impl Component for ShieldedPool {
-    type AppState = genesis::AppState;
+    type AppState = GenesisContent;
 
     #[instrument(name = "shielded_pool", skip(state, app_state))]
-    async fn init_chain<S: StateWrite>(mut state: S, app_state: &genesis::AppState) {
+    async fn init_chain<S: StateWrite>(mut state: S, app_state: &GenesisContent) {
         match app_state {
```

Comment:
I think something like this could work:

```rust
async fn init_chain<S: StateWrite>(mut state: S, app_state: Option<&GenesisContent>) {
    match app_state {
        None => { /* no-op */ },
        Some(app_state) => {
            /* bootstrapping */
        }
    }
    /* logic that is common to both checkpointed and full config genesis */
}
```

In most cases, this can be refactored even further since the checkpointing case is a no-op. Either way, the most important thing is that the common logic be executed, in particular in the staking component as we build the cometbft view of the validator set.

Reply:
This means we would only be able to perform checkpointing verification logic and health checks at the application-level `init_chain`, but that seems fine.

Reply:
The application state argument to the component's `init_chain` would be the component specific application state I imagine